server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -445,9 +445,8 @@ public void executeDfsPhase(ShardSearchRequest request, SearchShardTask task, Ac
     private DfsSearchResult executeDfsPhase(ShardSearchRequest request, SearchShardTask task) throws IOException {
         ReaderContext readerContext = createOrGetReaderContext(request);
         try (
-            Releasable scope = tracer.withScope(task);
             Releasable ignored = readerContext.markAsUsed(getKeepAlive(request));
-            SearchContext context = createContext(readerContext, request, task, ResultsType.DFS, false)
+            SearchContext context = createContext(readerContext, request, task, ResultsType.DFS, false, true)
```

Comment:
here we have the request at hand and could make additional decisions about parallelization based on its content